### ISSUE TITLE
bq274xx incorrect conversions

### ISSUE THREAD

**Issue by anthony32086** [zephyrproject-rtos/zephyr#54200]
**Describe the bug**
bq274xx channel conversions for the following channels are incorrect.
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/bq274xx/bq274xx.c#L195
```
SENSOR_CHAN_GAUGE_FULL_CHARGE_CAPACITY
SENSOR_CHAN_GAUGE_REMAINING_CHARGE_CAPACITY
SENSOR_CHAN_GAUGE_NOM_AVAIL_CAPACITY
SENSOR_CHAN_GAUGE_FULL_AVAIL_CAPACITY
SENSOR_CHAN_GAUGE_AVG_POWER
```
Per the [sensor.h](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/sensor.h#L155) header comments, the units for the above enums are to be returned in milli SI units however the values returned by the sensor and stored in `bq274xx->variable` are already in milli SI units. 
Reference Datasheet:
- [bq27411 page 9](https://www.ti.com/lit/gpn/bq27411-g1)
- [bq27421 page 10](https://www.ti.com/lit/gpn/bq27421-g1)

The bug currently results in a conversion to micro SI units (often 0 due to integer rounding) for val1 and val2 is off by a factor of 1000.

**Expected behavior**
Returned values in milli SI units in proper `sensor_value` format.

**Impact**
Returned values are off by a factor of 1000.

**Additional notes**
In the same function, the use of a floating point variable to format temp into zephyr's sensor_value format can have a negative impact on platforms that do not support hardware floating point.
Using fixed point would be more appropriate for this. eg:
```
static int bq274xx_channel_get(const struct device *dev,
			       enum sensor_channel chan,
			       struct sensor_value *val)
{
	struct bq274xx_data *bq274xx = dev->data;
	int32_t temp_q16;
	const int32_t temp_constant_q16 = 273.15f * (1 << 16);
...
	case SENSOR_CHAN_GAUGE_TEMP:
		temp_q16 = (bq274xx->internal_temperature * (1 << 16)) / 10;
		temp_q16 = temp_q16 - temp_constant_q16;
		val->val1 = temp_q16 / (1 << 16);
		val->val2 = ((temp_q16 % (1 << 16)) * 1000000) / (1 << 16);
		break;
...
```

**Comment by kartben** [comment]
@anthony32086 thanks for reporting this! Would you be interested in submitting a pull request to address this?

**Comment by nixward** [comment]
@fabiobaltieri do you know if this has been addressed?

**Comment by fabiobaltieri** [comment]
> @fabiobaltieri do you know if this has been addressed?

Does not look so. Wanna take this one on yourself?

**Comment by fabiobaltieri** [comment]
I'll self assign for now so it does not get lost.

**Comment by nixward** [comment]
Fixed by e3f5e1f
